src/chain.js

```diff
@@ -11,19 +11,25 @@ var map = require('./map');
  *
  * Dispatches to the `chain` method of the second argument, if present,
  * according to the [FantasyLand Chain spec](https://github.com/fantasyland/fantasy-land#chain).
+ * Treats the second argument as a FantasyLand Chain, when the the second argument is a function
+ * of the form `a -> x -> b`.
  *
  * @func
  * @memberOf R
  * @since v0.3.0
  * @category List
  * @sig Chain m => (a -> m b) -> m a -> m b
+ * @sig (x -> a) -> (a -> x -> b) -> (x -> b)
```

Comment:
This isn't strictly necessary.

```haskell
chain :: Chain m => (a -> m b) -> m a -> m b
```

Replace `Chain m => m` with `Function x`:

```haskell
chain :: (a -> Function x b) -> Function x a -> Function x b
```

Use `x -> y` sugar for `Function x y`:

```haskell
chain :: (a -> x -> b) -> (x -> a) -> (x -> b)
```

I'd prefer to let the general signature tell the story.